go-controller/pkg/ovn/default_network_controller.go

```diff
@@ -940,7 +947,9 @@ func (h *defaultNetworkControllerEventHandler) UpdateResource(oldObj, newObj int
 				klog.Infof("Node %s in remote zone %s needs interconnect zone sync up. Zone cluster changed: %v",
 					newNode.Name, util.GetNodeZone(newNode), zoneClusterChanged)
 			}
-			return h.oc.addUpdateRemoteNodeEvent(newNode, syncZoneIC)
+			_, syncHostNetAddrSet := h.oc.syncHostNetAddrSetFailed.Load(newNode.Name)
+			nodeParams = &nodeSyncs{false, false, false, false, false, syncZoneIC, syncHostNetAddrSet}
+			return h.oc.addUpdateRemoteNodeEvent(newNode, nodeParams)
```

Comment:
hmm this is making it complicated for me.. so currently we are changing too many things..
why didn't we take the approach of just calling `addIPToHostNetworkNamespaceAddrsSet` irrespective of whether node is local or remote straight from addR and updateR instead of nesting this one layer below into remote/local functions and then calling from both?? that way we don't need to change all these pieces to addUpdateRemoteNodeEvent.
Does that make sense?

Reply:
Thought of taking this approach to keep AddResource and UpdateResource clean. However it makes sense to keep the logic in AddResource/UpdateResource in order to keep it simple. Will take care of this in next push.